exp-player/addon/components/exp-alternation/component.js

```diff
@@ -0,0 +1,337 @@
+                            'type': {
+                                type: 'string'
+                            }
+                        }
+                    }
+                },
+            }
+        },
+        data: {
+            type: 'object',
+            properties: {
+                altOnLeft: {
+                    type: 'boolean'
+                },
+                videoId: {
+                    type: 'string'
+                }
+            }
+        }
+    },
+
+    actions: {
+
+        startVideo() {
+            // Set a timer to start the trial
+            $('#player-audio')[0].play();
+            var frame = this;
+            window.setTimeout(function(){
+                frame.set('completedAttn', true);
+                if (!frame.get('doingIntro')) {
+                    frame.startTrial();
+                }
+            }, frame.get('attnLength') * 1000);
+        },
+
+        endAudio() {
+            this.set('completedAudio', true);
+            if (!this.get('doingIntro')) {
+                    this.startTrial();
+            }
+        }
+
+    },
+
+
+    startTrial() {
+        var frame = this;
+        frame.presentTriangles( frame.settings.LshapesStart,
+                                        frame.settings.RshapesStart,
+                                        frame.settings.LsizeBaseStart,
+                                        frame.settings.RsizeBaseStart);
+        window.setTimeout(function(){
+            window.clearTimeout(frame.get('stimTimer'));
+            frame.clearTriangles();
+            frame.send('next');
+            }, frame.settings.trialLength * 1000);
+    },
+
+    sendTimeEvent(name, opts = {}) {
+        var streamTime = this.get('recorder') ? this.get('recorder').getTime() : null;
+
+        Ember.merge(opts, {
+            streamTime: streamTime,
+            videoId: this.get('videoId')
+        });
+        this.send('setTimeEvent', `exp-physics:${name}`, opts);
+    },
+
+    onFullscreen() {
+        if (this.get('isDestroyed')) {
+            return;
+        }
+        this._super(...arguments);
+        if (!this.checkFullscreen()) {
+            this.sendTimeEvent('leftFullscreen');
+            // TODO: if setting up pausing of experiment, also pause here.
+            //if (!this.get('isPaused')) {
+            //    this.pauseStudy();
+            //}
+        } else {
+            this.sendTimeEvent('enteredFullscreen');
+        }
+    },
+
+    stimTimer: null,
+
+    getRandomElement(arr) {
+      return arr[Math.floor(Math.random() * arr.length)];
+    },
+
+    getRandom(min, max) {
+      return Math.random() * (max - min) + min;
+    },
+
+    triangleBases: null,
+
+    drawTriangles(Lshape, LX, LY, LRot, LFlip, LSize, Rshape, RX, RY, RRot, RFlip, RSize) {
+
+        var leftTriangle = [this.triangleBases[Lshape],
+            '" transform=" translate(', LX, ', ', LY, ') ',
+            'translate(37.5, 56) rotate(', LRot, ') ',
```

Comment:
Yes, unfortunately this is where the issue with definining the shapes in the template comes in. It would make more sense to define the "skinny" and "fat" triangles in the template, then just do `<use xlink:href="#skinny" .../>` to set the transform values (translation, rotation, etc.), but on the Lookit site the xlink doesn't work/find the reference. OTOH this does keep all of the data relevant to the actual stimuli (range of X- and Y-motion, range of sizes, counterbalancing, coordinates) within didInsertElement in component.js.